### PR TITLE
IOS15 UNNotificationActionIcon support

### PR DESCRIPTION
**Pull request by azlindsay**
### Description of Changes

Add support for UNNotificationActionIcon on notification buttons, added with iOS 15

### Breaking Changes

-   None

### Release Checklist

Prepare:

-   [x] Detail any breaking changes. Breaking changes require a new major version number
-   [ ] Check `pod lib lint` passes

Bump versions in:

-   [x] `Sources/Kumulos.swift`
-   [x] `KumulosSdkSwift.podspec`
-   [x] `KumulosSdkSwiftExtension.podspec`
-   [x] `Build target version for plist`
-   [x] `README.md`

Release:

-   [ ] Squash and merge to master
-   [ ] Delete branch once merged
-   [ ] Create tag from master matching chosen version
-   [ ] Run `pod trunk push` to publish to CocoaPods

Post Release:

Update docs site with correct version number references

- [x] https://docs.kumulos.com/developer-guide/sdk-reference/swift/
- [x] https://docs.kumulos.com/getting-started/integrate-app/

Update changelog:

- [x] https://docs.kumulos.com/developer-guide/sdk-reference/swift/#changelog

### CODE REVIEW

File: Sources/Extension/KumulosNotificationService.swift [hunk 1]
```diff
@@ -87,13 +87,34 @@ public class KumulosNotificationService {
...
 
         return actionArray;
     }
+    
+    fileprivate class func getButtonIcon(button:AnyHashable:Any]) -> UNNotificationActionIcon? {
+        guard buttonDict["icon"] != nil else {
+            return nil
+        }
+        
+        let iconDict = buttonDict["icon"] as! [String:String]
```
Comment (cgwyllie) [inline Sources/Extension/KumulosNotificationService.swift:106]: Think we should prefer `as?` or `guard let` for these, you can stack multiple lets into one guard over multiple lines w/ commas I think

File: Sources/Extension/KumulosNotificationService.swift [hunk 2]
```diff
@@ -87,13 +87,34 @@ public class KumulosNotificationService {
...
+        
+        let iconType = iconDict["type"] as !String
+        let iconId = iconDict["iconId"] as !String
+        
+        if (iconType === "custom") {
+            //TODO - What if this doesnt exist? Catch exception -> return nil?
```
Comment (cgwyllie) [inline Sources/Extension/KumulosNotificationService.swift:112]: Would your TODO apply to the system image ones too? (different iOS versions shipping different icon libs?) Purely conjecture / not sure / will depend how up to date we keep the set.

File: Sources/Extension/KumulosNotificationService.swift [hunk 3]
```diff
@@ -87,13 +87,34 @@ public class KumulosNotificationService {
 
             let id = buttonDict["id"] as! String
             let text = buttonDict["text"] as! String
-
-            let action = UNNotificationAction(identifier: id, title: text, options: .foreground)
+            
+            let icon = getButtonIcon(button: buttonDict)
+            
+            //TODO - Necessary to guard this constructor with 15+ ?
+            let action = UNNotificationAction(identifier: id, title: text, options: .foreground, icon: icon)
```
Comment (cgwyllie) [inline Sources/Extension/KumulosNotificationService.swift:94]: Does this need an availability guard for iOS15+ ? (fall back to older constructor w/ no icons?)

Comment (azlindsay) [reply Sources/Extension/KumulosNotificationService.swift]: Line for adding actions + method for working out and returning icon guarded by ios15+ now.

File: Sources/Extension/KumulosNotificationService.swift (hunk 1 above)
Comment (azlindsay) [reply Sources/Extension/KumulosNotificationService.swift]: Done.

File: Sources/Extension/KumulosNotificationService.swift (hunk 2 above)
Comment (azlindsay) [reply Sources/Extension/KumulosNotificationService.swift:112]: In testing this doesnt throw exceptions, just returns a UNNotificationActionIcon that eventually isnt drawn, button is still presented and clicks handled normally.

**Pull request merged**